axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetLineMoveServiceImpl.java

```diff
@@ -510,53 +508,58 @@ public void generateDisposalMove(
+            || transferredReason == FixedAssetRepository.TRANSFERED_REASON_PARTIAL_CESSION) {
+          if (fixedAsset.getFixedAssetCategory().getRealisedAssetsValueAccount() == null) {
+            throw new AxelorException(
+                TraceBackRepository.CATEGORY_MISSING_FIELD,
+                I18n.get(
+                    IExceptionMessage
+                        .IMMO_FIXED_ASSET_GENERATE_DISPOSAL_MOVE_CATEGORY_ACCOUNTS_MISSING),
+                I18n.get("Realised Assets Value Account"));
+          }
+          chargeAccount = fixedAsset.getFixedAssetCategory().getRealisedAssetsValueAccount();
+        } else {
+          chargeAccount = fixedAsset.getFixedAssetCategory().getChargeAccount();
         }
-        chargeAccount = fixedAsset.getFixedAssetCategory().getRealisedAssetsValueAccount();
-      } else {
-        chargeAccount = fixedAsset.getFixedAssetCategory().getChargeAccount();
-      }
-      MoveLine chargeAccountDebitMoveLine =
-          moveLineCreateService.createMoveLine(
-              move,
-              partner,
-              chargeAccount,
-              chargeAmount,
-              true,
-              disposalDate,
-              1,
-              origin,
-              fixedAsset.getName());
-      moveLines.add(chargeAccountDebitMoveLine);
-
-      this.addAnalyticToMoveLine(
-          fixedAsset.getAnalyticDistributionTemplate(), chargeAccountDebitMoveLine);
-
-      if (cumulativeDepreciationAmount != null && cumulativeDepreciationAmount.signum() > 0) {
-
-        MoveLine deprecationAccountDebitMoveLine =
+        MoveLine chargeAccountDebitMoveLine =
             moveLineCreateService.createMoveLine(
                 move,
                 partner,
-                depreciationAccount,
-                cumulativeDepreciationAmount,
+                chargeAccount,
+                chargeAmount,
                 true,
                 disposalDate,
                 1,
                 origin,
                 fixedAsset.getName());
-        moveLines.add(deprecationAccountDebitMoveLine);
+        moveLines.add(chargeAccountDebitMoveLine);
+
         this.addAnalyticToMoveLine(
-            fixedAsset.getAnalyticDistributionTemplate(), deprecationAccountDebitMoveLine);
+            fixedAsset.getAnalyticDistributionTemplate(), chargeAccountDebitMoveLine);
       }
+      if (chargeAmount.signum() == 0
+          && (cumulativeDepreciationAmount == null
+              || (cumulativeDepreciationAmount != null
+                  && cumulativeDepreciationAmount.signum() > 0))) {
+        cumulativeDepreciationAmount = fixedAsset.getGrossValue();
+      }
+      MoveLine deprecationAccountDebitMoveLine =
+          moveLineCreateService.createMoveLine(
+              move,
+              partner,
+              depreciationAccount,
+              cumulativeDepreciationAmount,
+              true,
+              disposalDate,
+              1,
```

Comment:
Shouldn't the ref be 2 instead of 1 since `chargeAccountDebitMoveLine` has already been created for that move ? And then set it as 3 for the following `creditMoveLine`. Maybe using a counter can be nice